### PR TITLE
Get previous Comments from ReadMe File Feature

### PR DESCRIPTION
**Pull request by sahandfarahani**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -49,10 +44,42 @@ const App: React.FC = () => {
     const owner = pathArray[3];
     const repo = pathArray[4];
     setRepoName(repo);
+    // await getOldTree(owner, repo);
```
Comment (ralph-dev) [inline src/App.tsx:47]: remove old code

File: src/App.tsx [hunk 2]
```diff
@@ -49,10 +44,42 @@ const App: React.FC = () => {
...
+    // if (oldTree) {
     await makeRequest(owner, repo);
+    // }
   };
 
   const makeRequest = async (owner: String, repo: String) => {
+    let oldTree: oldTree[] | null = null;
+
+    // Previous Comments
```
Comment (ralph-dev) [inline src/App.tsx:56]: remove useless comment

File: src/App.tsx [hunk 3]
```diff
@@ -49,10 +44,42 @@ const App: React.FC = () => {
...
+        const file = resJSON[key];
+        if (file["path"] === "README.md") {
+          const SHA = file["sha"];
+          const blobs = await fetch(
+            `https://api.github.com/repos/${owner}/${repo}/git/blobs/${SHA}`
+          );
+          const blobsJSON = await blobs.json();
+          const decodedBlobs = atob(blobsJSON["content"]);
+          const haveComments = decodedBlobs.match(
+            /((\[.+)\]\(\.\/.+\)\s+# .+)/g
```
Comment (ralph-dev) [inline src/App.tsx:73]: abstract this regex to make more sense?
const commentsExistRegex = /rwerewr/g etc

File: src/App.tsx [hunk 4]
```diff
@@ -49,10 +44,42 @@ const App: React.FC = () => {
...
+    // Previous Comments
+    try {
+      const res = await fetch(
+        `https://api.github.com/repos/${owner}/${repo}/contents`
+      );
+      const resJSON = await res.json();
+
+      for (const key in resJSON) {
+        const file = resJSON[key];
+        if (file["path"] === "README.md") {
```
Comment (ralph-dev) [inline src/App.tsx:65]: const README = "readme.md" etc

File: src/App.tsx (hunk 2 above)
Comment (sahandfarahani) [reply src/App.tsx:56]: @ralph-dev all segments in makeRequest has a comment above them... should I remove all of them or just Previous Comment?

Comment (ralph-dev) [reply src/App.tsx:56]: Previous comment provides no value, thus it should be removed. This rule applies to any comment that provides no value

File: src/utils/getPreviousTree.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  comment: string | undefined;
+}
+
+export const getPreviousTree = (haveComments: string[] | null): oldTree[] => {
+  if (haveComments === null) {
+    return [];
+  }
+  const haveCommentsArray: oldTree[] = [];
+  haveComments.map((line, key) => {
+    const firstHalf = line.match(/(\(\.\/.+\))/g)?.toString();
```
Comment (ralph-dev) [inline src/utils/getPreviousTree.ts:17]: Have a const for the regex so we know in English what you are pattern matching

**Comment by ralph-dev** [comment]
Make sure to delete the branch

**Pull request merged**